### ISSUE TITLE
Provide a Foundation-based HTTP Client

### ISSUE THREAD

**Issue by ameter** [awslabs/aws-sdk-swift#817]
### Describe the feature

The CRT-based HTTP client does not respect system proxy settings, which prevents service calls from being observable via standard proxy-based network protocol analyzers.  (e.g. Charles Proxy, Proxyman, etc).

### Use Case

Being able to observe network traffic is common practice when debugging and testing apps.  This is also often done to validate behavior during security testing.

### Proposed Solution

Provide a standard Foundation-based HTTP client with the SDK.  Assuming this will not be the default HTTP client, also provide documentation for customers to swap in this client for testing.

For example, something like this:

```
import Foundation
import ClientRuntime

struct FoundationHttpClient : HttpClientEngine {
    func execute(request: ClientRuntime.SdkHttpRequest) async throws -> ClientRuntime.HttpResponse {
        let url = request.endpoint.url!
        var urlRequest = URLRequest(url: url)
        urlRequest.httpMethod = request.method.rawValue
        
        for header in request.headers.headers {
            for value in header.value {
                urlRequest.addValue(value, forHTTPHeaderField: header.name)
            }
        }
    
        switch request.body {
        case .data(let data):
            urlRequest.httpBody = data
        case .stream(let stream):
            urlRequest.httpBody = stream.toBytes().toData()
        case .none:
            break
        }
        let (data, response) = try await URLSession.shared.data(for: urlRequest)
        let httpResponse = response as! HTTPURLResponse
        let statusCode = HttpStatusCode(rawValue: httpResponse.statusCode)!
        
        var headers = Headers()
        for header in httpResponse.allHeaderFields {
            if let key = header.key as? String, let value = header.value as? String {
                headers.add(name: key, value: value)
            }

            if let key = header.key as? String, let values = header.value as? [String] {
                headers.add(name: key, values: values)
            }
        }
        return HttpResponse(headers: headers,
                            body: HttpBody.stream(ByteStream.from(data: data)),
                            statusCode: statusCode)
    }
    
    func close() async {
    }
}
```

Could be used in a service client with something like the following (modifying the config before instantiating the client):
```
configuration.httpClientEngine = FoundationHttpClient()
```

### Other Information

There is a reference to [this issue with other third party networking libraries](https://docs.proxyman.io/troubleshooting/couldnt-see-any-requests-from-3rd-party-network-libraries) in the Proxyman docs.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change
```[tasklist]
### Tasks
```

**Comment by ganeshnj** [comment]
We must do #695 first and then tackle this.

**Comment by jbelkins** [comment]
This work is a prerequisite for Apple Watch platform support.

cc: @smivij

**Comment by smivij** [comment]
As discussed, once complete we will add CI tasks to run on Apple Watch

**Comment by jbelkins** [comment]
Shipped in AWS SDK for Swift 0.34.0
